### PR TITLE
Add documentation on Kerberos configuration

### LINKED ISSUE #7295: [DOC] Add information regarding Kerberos integration
**What do you want to do?**
 
- [x] Add new documentation

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 Currently OpenSearch security plugin supports Kerberos as Authentication backend. However, it is not listed on the relevant section: https://opensearch.org/docs/latest/security/authentication-backends/authc-index/

So, let's add it. Here's an example working config:
```
...
      "authc": {
        "basic_internal_auth_domain": {
          "http_enabled": true,
          "transport_enabled": true,
          "order": 1,
          "http_authenticator": {
            "challenge": false,
            "type": "basic",
            "config": {}
          },
          "authentication_backend": {
            "type": "intern",
            "config": {}
          },
          "description": "Authenticate via HTTP Basic against internal users database"
        },
        "kerberos_auth_domain": {
          "http_enabled": true,
          "transport_enabled": true,
          "order": 2,
          "http_authenticator": {
            "challenge": true,
            "type": "kerberos",
            "config": {
              "krb_debug": false,
              "strip_realm_from_principal": true
            }
          },
          "authentication_backend": {
            "type": "noop",
            "config": {}
          },
          "description": "Authenticate via Kerberos"
        }
...

### PR DESCRIPTION
**Pull request by spapadop**
### Description
Kerberos is currently supported on OpenSearch as authentication backend, but it is not documented.

### Issues Resolved
Close #7295 

### Version
all

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by spapadop** [comment]
@hdhalter  ready :)

**Review by derek-ho — APPROVED** [review]
Changes look good to me. @cwperks maybe can provide another set of eyes here. It seems like this is mostly moving commented out docs out.

File: _security/authentication-backends/kerberos.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+As with [HTTP Basic Authentication]({{site.url}}{{site.baseurl}}/security/authentication-backends/basic-authc/), this flag determines how the Security plugin should react when no `Authorization` header is found in the HTTP request or if this header does not equal `negotiate`.
+
+If set to `true`, the Security plugin sends a response with status code 401 and a `WWW-Authenticate` header set to `negotiate`. This tells the client (browser) to resend the request with the `Authorization` header set. If set to `false`, the Security plugin cannot extract the credentials from the request, and authentication fails. Setting `challenge` to `false` thus makes sense only if the Kerberos credentials are sent in the initial request.
+
+Name | Description
+:--- | :---
+`krb_debug` | As the name implies, setting it to `true` will output Kerberos-specific debugging messages to `stdout`. Use this setting if you encounter problems with your Kerberos integration.
```
Comment (cwperks) [inline _security/authentication-backends/kerberos.md:60]: nit: Include what the default values are in the description: i.e. `Default: false`

File: _security/configuration/configuration.md [hunk 2]
```diff
@@ -162,65 +163,4 @@ To learn about configuring the authentication backends, see the [Authentication
 * [Active Directory and LDAP]({{site.url}}{{site.baseurl}}/security/authentication-backends/ldap/)
 * [Proxy-based authentication]({{site.url}}{{site.baseurl}}/security/authentication-backends/proxy/)
 * [Client certificate authentication]({{site.url}}{{site.baseurl}}/security/authentication-backends/client-auth/)
-
-
-<!--- Remvoving Kerberos documentation until issue #907 is resolved.
```
Comment (cwperks) [inline _security/configuration/configuration.md:167]: I see that https://github.com/opensearch-project/security-dashboards-plugin/issues/907 is still open. While we don't have automated functional tests in the security-dashboards-plugin to verify kerberos, has it been validated through a demo configuration?

File: _security/authentication-backends/kerberos.md (hunk 1 above)
Comment (spapadop) [reply _security/authentication-backends/kerberos.md]: good point! added

File: _security/configuration/configuration.md (hunk 2 above)
Comment (spapadop) [reply _security/configuration/configuration.md:167]: I assume that kerberos is still not supported on OpenSearch Dashboards, but this documentation is rather about OpenSearch, where kerberos is supported. We've been using it successfully on all our clusters already from the OpenDistro era.

**Comment by spapadop** [comment]
@Naarcha-AWS thanks for the commit, changes seem good! But the DCO check if failing ever since, could you please sign accordingly?

**Review by Naarcha-AWS — APPROVED** [review]

**Pull request merged**